src/twisted/internet/cfreactor.py

```diff
@@ -372,22 +372,85 @@ def _moveCallLaterSooner(self, tple):
         self._scheduleSimulate()
         return result
 
+    def startRunning(self, installSignalHandlers: bool = True) -> None:
+        """
+        Start running the reactor, then kick off the timer that advances
+        Twisted's clock to keep pace with CFRunLoop's.
+        """
+        super().startRunning(installSignalHandlers)
+        # We must force the simulator to run because in addition to scheduling
+        # timed calls, super().startRunning also fires startup events which may
+        # crash the reactor.
```

Comment:
I don't quite follow this explanation.  I tried to expand some of the implications here but I don't think I got somewhere coherent:

```
        # Prior to this point (maybe because of the super().startRunning)                                                                                   
        # arbitrary application code may have run.  This may include code that                                                                              
        # added event sources (timers or file descriptors) to the reactor.  If                                                                              
        # so, either _scheduleSimulate will already have run and added a Timer                                                                              
        # to the CFRunLoop or _watchFD will have added a Source.  Either way,                                                                               
        # there is work for CFRunLoop to do and things will work well.                                                                                      
        #                                                                                                                                                   
        # If neither of those things happened then there is no work for the                                                                                 
        # CFRunLoop to do.                                                                                                                                  
        #                                                                                                                                                   
        # ... so what?                                                                                                                                      
```

Reply:
This is … directionally correct, so I'll try to do an update that addresses some of these problems

Reply:
Aaand upon improving the verbiage I realized the previous version was wrong anyway. Oops! Hopefully the new version is correct.